frontend/src/main/scala/bloop/data/SettingsMapper.scala

```diff
@@ -0,0 +1,106 @@
+package bloop.data
+import bloop.engine.BuildLoader
+import bloop.logging.Logger
+import bloop.logging.DebugFilter
+import java.nio.charset.StandardCharsets
+import bloop.config.ConfigEncoderDecoders
+import io.circe.Printer
+import io.circe.Json
+import io.circe.parser
+import bloop.io.AbsolutePath
+import java.nio.file.Files
+import bloop.config.Config.Settings
+import io.circe.JsonObject
+import _root_.bloop.config.Config
+import bloop.DependencyResolution
+import scala.util.Try
+
+object SettingsMapper {
+
+  /**
+    * @param settingsPath Path to the json settings file.
+    * @param logger The logger that collects messages about loading the settings.
+    * @return Settings if the file exists.
+    */
+  def fromFile(settingsPath: AbsolutePath, logger: Logger): Option[Settings] = {
```

Comment:
Don't want to remove Settings always , because `None` will mean we don't want to add SemanticDB plugin. Will rename to apply.

Reply:
If you rename to apply, let's return `Settings`, it's the convention. I propose instead of checking for `None`, you check for `settings == DefaultSettings`. It's effectively the same thing and IMO makes sense.

Reply:
Not sure I understand, what would `settings == DefaultSettings` give us? `None` would mean that we don't have any settings, so we shouldn't apply them. If default settings was just `Settings(None)` then how would it be different from `Settings(None)` that means we should get `release.latest`?

Reply:
Nevermind :smile: After having a look at the code again, I propose we keep it as is. `Settings.fromFile` is good. I was just mentioning that the convention I use for `apply` is that it always returns the same type of the companion class, returning `Option[Settings]` looks weird.